### PR TITLE
Patch for staged files only

### LINKED ISSUE #3716: Patch for Stages Files only
We are working on a huge repository with hundreds or thousands of changed objects sometimes and multiple developers working on the same objects in certain cases.
I now know that abapGit already has hunk functionality in form of the 'Patch' functionality on the staging screen; howvere, I noticed that pressing Patch on the stage screen always lists all changed objects on the following patch screen, which is not only unwieldy and convoluted, but also takes a longtime to prepare, which takes a lot of waiting time (e.g. having abapGit render diffs for 500 changes objects even though I know I have only changed 1 is rather annoying)
I've checked the coding and have seen that the ZCL_ABAPGIT_GUI_PAGE_PATCH could probably easily be adapted to support another case (currently only single file, single object or whole repo are supported) and render patches for a given set of objects and the ZCL_ABAPGIT_GUI_PAGE_STAGE would be able to supply this set of objects based on the staged state of the objects.
However, while pressing "Commit selected" si supplied with the POST data from which the selected objects can be determined, pressing Patch on the same page does not receive any POST data.
Could you provide me with some hints, where I would have to look to get the POST data of the selected objects when Patch is chosen?
2020-08-06_12h02_10

### PR DESCRIPTION
**Pull request by christianguenter2**
closes #3716

### CODE REVIEW

File: src/ui/zabapgit_js_common.w3mi.data.js [hunk 1]
```diff
@@ -1288,12 +1295,20 @@ function Hotkeys(oKeyMap){
     // the hotkey execution
     this.oKeyMap[sKey] = function(oEvent) {
 
+      var helper = (window.gHelper || {});
```
Comment (sbcgua) [inline src/ui/zabapgit_js_common.w3mi.data.js:1298]: Could you pls clarify this code, seems to not used directly anywhere

File: src/ui/zabapgit_js_common.w3mi.data.js [hunk 2]
```diff
@@ -1288,12 +1295,20 @@ function Hotkeys(oKeyMap){
...
+
       // We have either a js function on this
       if (this[action]) {
         this[action].call(this);
         return;
       }
 
+      // Or a method of the helper object
+      if (helper[action]){
```
Comment (sbcgua) [inline src/ui/zabapgit_js_common.w3mi.data.js:1307]: And this part of it

**Comment by sbcgua** [comment]
Oops. Pressed submit before the comment. So comminting here.

The code works, just like intended. 2 points to check:
- could you pls clarify the code above ? Is it related to this PR ?
- one minor bug:  if I choose files, go to patch (1 file displayed) and then press Refresh (at the file level) - all files are redisplayed

But otherwise everything looks fine and works. :+1:

**Comment by christianguenter2** [comment]
@sbcgua thanks for your review

Regarding your questions

- `could you pls clarify the pieces above ? Is it related to this PR ?`

This is needed because the patch button on stage page changed from `sapevent` to `onclick`. And thefefore hotkey handling has to be different. With this change it's possible to hotkey JS methods of the gHelper object which exists for the stage page. This was not possible before. And I need that hotkey desperately ;-) I added clarifying comments in the code.

Hotkey definition
https://github.com/christianguenter2/abapGit/blob/06e86081550217158d324cc06a88c8685bb50842/src/ui/zcl_abapgit_gui_page_stage.clas.abap#L531

gHelper definition
https://github.com/christianguenter2/abapGit/blob/06e86081550217158d324cc06a88c8685bb50842/src/ui/zcl_abapgit_gui_page_stage.clas.abap#L718-L722

- ` one minor bug: if I choose files, go to patch (1 file displayed) and then press Refresh (at the file level) - all files are redisplayed`

This bug and another one (Diff  and Patch buttons are not visible when there are only remote changes) are fixed

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**